docs/site/Deploying_to_ibm_cloud_kubernetes.md

```diff
@@ -0,0 +1,707 @@
+- [IBM Cloud CLI, Container registry CLI, etc](https://cloud.ibm.com/docs/cli/index.html#overview)
+- [Kubernetes CLI (kubectl)](https://kubernetes.io/docs/reference/kubectl/overview/)
+- LoopBack 4 CLI
+
+Let's install the LoopBack 4 CLI :
+
+```
+npm i -g @loopback/cli
+```
+
+After you have scaffolded a LoopBack4 application, you’ll get a default endpoint /ping. For simplicity sake, we’ll use this.
+
+## Step 1: Scaffold LoopBack 4 app
+
+Run ````lb4 app```` command.
+
+````
+$ lb4 app
+? Project name: deploy-test
+? Project description: loopback example
+? Project root directory: deploy-test
+? Application class name: DeployTestApplication
+? Select features to enable in the project Enable tslint, Enable prettier, Enable mocha, Enable loopback
+Build, Enable vscode, Enable repositories, Enable services
+...
+Application deploy-test was created in deploy-test.
+Next steps:
+$ cd deploy-test
+$ npm start
+````
+
+Navigate to the main directory of your project
+
+````
+cd deploy-test
+````
+
+Let's **not** run ````npm start```` yet, we have some changes to make to certain files.
+
+## Step 2: Modify deploy-test/index.js
+
+It is important for the host to be ````'0.0.0.0'```` and the port to be consistent in several places. We will use a port of ````8080````.
+
+````
+  const config = {
+    rest: {
+      port: 8080,
+      host: '0.0.0.0',
+      openApiSpec: {
+        // useful when used with OASGraph to locate your application
+        setServersFromRequest: true,
+      },
+    },
+  };
+````
+
+## Step 3: Modify deploy-test/src/index.ts
+
+It is important for the host to be ````'0.0.0.0'```` and the port to be consistent in several places. We will use a port of ````8080````.
+
+
+````
+import {DeployTestApplication} from './application';
+import {ApplicationConfig} from '@loopback/core';
+export {DeployTestApplication};
+
+export async function main(options: ApplicationConfig = {}) {
+
+  // --------- ADD THIS SNIPPET -------------
+if (!options) options = {};
+if (!options.rest) options.rest = {};
+options.rest.port = 8080;
+options.rest.host = "0.0.0.0";
+// ----------------------------------------
+
+const app = new DeployTestApplication(options);
+  await app.boot();
+  await app.start();
+
+  const url = app.restServer.url;
+  console.log(`Server is running at ${url}`);
+  console.log(`Try ${url}/ping`);
+
+  return app;
+}
+````
+
+## Step 4: Add .dockerignore file
+
+At project root, create a file called ````.dockerignore```` with the following content:
+
+````
+node_modules
+npm-debug.log
+````
+
+## Step 5: Update package.json
+
+Before we deploy to the cloud, go to package.json at the root of the project directory, and remove the following line:
+
+````
+"prestart": "npm run build",
+````
+
+## Step 6: Build the application
+
+In a command window, navigate the main directory of your project
+and type :
+
+````
+npm run build
+````
+
+Make sure there are no errors before continuing to the next step.
+
```

Comment:
Why do you want remove `prestart` and manually run `npm run build` before `npm start`?

Reply:
This was a step suggested in 

https://medium.com/loopback/deploy-your-first-loopback4-application-to-ibm-cloud-2e671ac06b6c 

published by @dhmlau .

I figured there was a good reason to do this, so I copied the instruction.